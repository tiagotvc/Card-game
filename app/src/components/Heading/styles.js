import styled, { css } from 'styled-components';

const textSize = {
    small: () => css`
      font-size: 10px ;
    `,
    medium: () => css`
      font-size: 15px ;
    `,
    big: () => css`
      font-size: 22px ;
    `,
    huge: () => css`
      font-size: 35px ;
    `,
  };
  
  const textCase = (uppercase) => css`
    text-transform: ${uppercase ? 'uppercase' : 'none'};
  `;

export const Text = styled.h1`
    ${({size, uppercase, color }) => css`
    color: ${color};
    ${textSize[size]};  ${textSize[size]};
    ${textCase(uppercase)};
    `}
`;