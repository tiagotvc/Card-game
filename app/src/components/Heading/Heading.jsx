import React from 'react';
import P from 'prop-types';
import * as Styled from './styles';


export const Heading = ({
    size = 'big',
    uppercase = false,
    children, 
    as = 'h1',
    color = 'black'
}) => {
    return (
        <Styled.Text
            size={size}
            uppercase={uppercase}
            as={as}
            color={color
        }>
            {children}
        </Styled.Text>
    );
};

Heading.propTypes = {
    children: P.node.isRequired,
    as: P.oneOf(['h1', 'h2', 'h3', 'h4', 'h5', 'h6']).isRequired,
    size: P.oneOf(['small', 'medium', 'big', 'huge']).isRequired,
    uppercase: P.bool.isRequired,
    color: P.node.isRequired
  };

  //prevent jest error
export default Heading;