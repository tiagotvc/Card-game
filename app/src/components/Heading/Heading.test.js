import React from 'react';
import Heading from './Heading';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

describe('Heading ', () => {
    it('renders correctly', () => {
      const tree = renderer
      .create(<Heading>Not</Heading>)
      .toJSON();
    expect(tree).toMatchSnapshot();
    });
})