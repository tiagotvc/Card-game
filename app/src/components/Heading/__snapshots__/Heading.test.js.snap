// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading  renders correctly 1`] = `
<h1
  className=""
  color="black"
  size="big"
>
  Not
</h1>
`;
